server/controllers/stock/core.js

```diff
@@ -508,6 +510,40 @@ async function getStockConsumptionAverage(periodId, periodDate, monthAverageCons
     GROUP BY i.uuid, d.uuid;
   `;
 
+  // the value (w.quantity/w.days) is the CMM
+  // is mentionned here as quantity to fit into the expected values
+  // for returned values
+  const queryConsumptionByDays = `
+  SELECT
+    w.consumption AS counted_consumption,
+    w.quantity AS consumed_quantity,
+    ROUND(w.quantity/w.days) AS quantity,
+    w.depot_text, w.code, w.text, w.days,
+    BUID(w.depot_uuid) depot_uuid, BUID(w.inventory_uuid) uuid
+  FROM 
+  (
+    SELECT
+      z.depot_text, z.text, SUM(IF(z.date, 1, 0)) days, SUM(z.consumption) consumption, SUM(z.quantity) quantity,
+      z.depot_uuid, z.inventory_uuid, z.code
+    FROM (
+      SELECT 
+        count(*) consumption, SUM(sm.quantity) quantity,  sm.date,
+        i.uuid inventory_uuid, i.text, i.code,
+        d.uuid depot_uuid, d.text AS depot_text
+      FROM stock_movement sm 
+      JOIN lot l ON l.uuid = sm.lot_uuid
+      JOIN inventory i ON i.uuid = l.inventory_uuid
+      JOIN depot d ON d.uuid = sm.depot_uuid
+      WHERE 
+        (DATE(sm.date) BETWEEN ? AND ?) AND flux_id IN (${flux.TO_PATIENT}, ${flux.TO_SERVICE})
+      GROUP BY d.uuid, i.uuid, sm.date
+      HAVING consumption <> 0
```

Comment:
This is better than the previous version (considering the entire month), but we still want to only consider days where quantity in stock > 0, rather than consumption <> 0.  

However, if you intend this to be a step in the right direction, I can see this being a small improvement to the calculation.  What do you think?

Reply:
We'll address this in another PR.  This is a significant enough improvement that we should merge it, in my view.

Reply:
Yes, this is what we want.  We want to count days with no movements, but with quantity in stock.  Those days will have a CMM of 0.  Will `quantity > 0` do that for us?